packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -1157,65 +1159,62 @@ function updateDeferredValue<T>(
   const [prevValue, setValue] = updateState(value);
   updateEffect(
     () => {
-      Scheduler.unstable_next(() => {
-        const previousConfig = ReactCurrentBatchConfig.suspense;
-        ReactCurrentBatchConfig.suspense = config === undefined ? null : config;
-        try {
-          setValue(value);
-        } finally {
-          ReactCurrentBatchConfig.suspense = previousConfig;
-        }
-      });
+      const previousConfig = ReactCurrentBatchConfig.suspense;
+      ReactCurrentBatchConfig.suspense = config === undefined ? null : config;
+      try {
+        setValue(value);
+      } finally {
+        ReactCurrentBatchConfig.suspense = previousConfig;
+      }
     },
     [value, config],
   );
   return prevValue;
 }
 
+function startTransition(setPending, config, callback) {
+  const priorityLevel = getCurrentPriorityLevel();
+  runWithPriority(
```

Comment:
I feel like we do this in a couple of internal places which seems like a very heavy weight way to do this. Going through the scheduler to do this. This adds to my suspicion that reading priorities from scheduler is the wrong model. It's not that scheduler dictates the priorities but it sometimes can be used as a default.

Reply:
I agree, it's on my list of things to consider during expiration times refactor. Didn't want to couple it to this bugfix.